File: scripts/import_excel.py
import openpyxl
import os
import sys

# Ensure UTF-8 output encoding for Windows terminal
sys.stdout.reconfigure(encoding='utf-8')

EXCEL_FILE = 'Costaff_Master Tracker_V.1.xlsx'

def main():
    print(f"Starting Excel Data Importer & Inspector for: {EXCEL_FILE}")
    if not os.path.exists(EXCEL_FILE):
        print(f"File not found: {EXCEL_FILE}")
        sys.exit(1)

    size_mb = os.path.getsize(EXCEL_FILE) / (1024 * 1024)
    print(f"File Size: {size_mb:.2f} MB")

    print("Loading workbook sheet names...")
    wb = openpyxl.load_workbook(EXCEL_FILE, read_only=True)
    sheet_names = wb.sheetnames
    print(f"Discovered {len(sheet_names)} Sheets: {sheet_names}")

    print("\nInspecting Row & Column Counts per Sheet:")
    for sheet_name in sheet_names:
        sheet = wb[sheet_name]
        max_r = sheet.max_row
        max_c = sheet.max_column
        print(f"  - [{sheet_name}]: {max_r} rows x {max_c} columns")

    print("\nExcel Data Inspector Completed Successfully!")

if __name__ == '__main__':
    main()

File: scripts/importExcel.js
const fs = require('fs');
const path = require('path');
const xlsx = require('xlsx');

const EXCEL_FILE = path.join(__dirname, '../Costaff_Master Tracker_V.1.xlsx');

function main() {
  console.log(`🚀 Starting Excel Data Importer for: ${EXCEL_FILE}`);

  if (!fs.existsSync(EXCEL_FILE)) {
    console.error(`❌ File not found: ${EXCEL_FILE}`);
    process.exit(1);
  }

  const stats = fs.statSync(EXCEL_FILE);
  console.log(`📦 File Size: ${(stats.size / (1024 * 1024)).toFixed(2)} MB`);

  console.log('⏳ Reading workbook sheet names...');
  const workbook = xlsx.readFile(EXCEL_FILE, { bookSheets: true });
  console.log('📋 Discovered Sheets:', workbook.SheetNames);

  const sheetsToProcess = [
    'Sheet1',
    'Client_Staffing Log',
    'TA_Allocation Log',
    'TA_Daily Call Log',
    'Interview Sheet',
    'Onboarding Sheet',
    'AM_TA_Dashboard',
    'Monthly Targets Achieved',
    'Do_NOT_Delete'
  ];

  console.log('\n🔍 Inspecting Row Counts per Sheet:');
  const fullWb = xlsx.readFile(EXCEL_FILE, { cellDates: true, sheetRows: 50000 });

  for (const sheetName of sheetsToProcess) {
    if (fullWb.Sheets[sheetName]) {
      const sheet = fullWb.Sheets[sheetName];
      const data = xlsx.utils.sheet_to_json(sheet, { header: 1 });
      console.log(`  ✅ [${sheetName}]: ${data.length} total rows, ${data[0] ? data[0].length : 0} columns`);
    } else {
      console.log(`  ⚠️ [${sheetName}]: Sheet missing in workbook!`);
    }
  }

  console.log('\n🎉 Excel Data Validation Completed Successfully!');
}

main();
